core/src/test/groovy/io/micronaut/core/util/ObjectUtilsSpec.groovy

```diff
@@ -21,4 +22,34 @@ class ObjectUtilsSpec extends Specification {
             o3 << ["abc", null, "xyz", null]
     }
 
+    @Unroll("ObjectUtils.coerceToBoolean with argument #obj returns #expected")
+    def "ObjectUtils::coerceToBoolean"(boolean expected, Object obj) {
+        expect:
+        expected == ObjectUtils.coerceToBoolean(obj)
+        where:
+        expected | obj
+        false    | null
+        false    | Boolean.FALSE
+        true     | Boolean.TRUE
+        true     | "string"
+        false    | ""
+        false    | 0L
+        false    | new BigDecimal("0.0")
+        false    | 0
+        false    | 0.0f
+        true     | 1L
+        true     | new BigDecimal("0.1")
+        true     | 1
+        false    | -1
```

Comment:
In Apache Groovy a negative number evaluates to `true`. Not sure if we should change this to return true for negative numbers as well.

Reply:
yeah I think the logic for numbers is questionable. We should return `true` as well